engine/engine.go

```diff
@@ -148,6 +152,10 @@ func (l distributedEngine) NewRangeQuery(q storage.Queryable, opts *promql.Query
 }
 
 func New(opts Opts) *compatibilityEngine {
+	return Create(opts, promql.NewEngine(opts.EngineOpts))
+}
+
+func Create(opts Opts, engine *promql.Engine) *compatibilityEngine {
```

Comment:
Does the `engine` param need to be a `promql.Engine` type? I guess we can make `prom` in `compatibilityEngine` a `v1.QueryEngine` type. Because when thanos creates a prometheus engine it gets type casted into `v1.QueryEngine`.